### ISSUE TITLE
Surface reactions

### ISSUE THREAD

**Issue by bslakman** [ReactionMechanismGenerator/RMG-Py#393]
I am trying to figure out if adding simple adsorption/desorption surface reactions to RMG is feasible at this point or not. I have been brainstorming about how we could do it and would be grateful if anyone had something to add to the discussion. 

A few options/approaches to this...
1) Whether or not the species is adsorbed or not can be a species property, just like number of lone pairs, bonds, radicals. We can define "actions" which change this property during a reaction, just like LOSE_PAIR, INCREMENT_BOND etc. But in this case, without explicitly accounting for the surface, I'm not sure how to represent that the reaction rates are proportional to the density of sites. And, not all sites are the same.
2) So that leads me to think we should represent a surface as a reactant with a "concentration". That way we can write a rate law that is dependent on the density of sites. The question is, how detailed is our representation of the surface? Do we have different types of sites with general properties, the way we might represent a new atom type? Or do we represent a portion of the surface with its full molecular structure?
3) Presence of a surface is just another "correction" to thermo, kinetics, etc. Like a solvation correction. This is attractive to me because I can apply concepts from what I've already done, but again since the rate doesn't just depend on what the surface, but the number of sites, I'm not sure how feasible it is.

Any comments are appreciated!

**Comment by connie** [comment]
From just reading this I think Option 2 is most intuitive.  In surface reactions it seems key to keep track of the number of sites as a sort of concentration.  The kinetics of the surface site with each type of gas molecule dictates what proportion of the sites are free.  Then you can sort of consider bound sites to be a diff species (spec A-B)  that has some binding energy of dissociation that is incorporated into its thermo? 

So basically one would end up with a family dictating the binding kinetics and you can build up thermo groups that incorporate the effects of the binding energy on the bound 'surfaceA-gasB' configuration.  What do you think?

**Comment by bslakman** [comment]
I think option 2 makes sense. I'm just unclear about how to represent "surfaceA" or "surfaceA-gasB" in our current framework of RMG. Whether A is its own species (and how specific to make that), or a property of gasB that gets turned on or off if it's bound or unbound.

**Comment by enochd** [comment]
I think it would be helpful to have list of requirements that must be met for implementing this. For example, rates are dependent on sticking coefficients, site density, the site type itself, at the least. Every open site is considered a distinct species. New databases would need to be introduced, and  group additivity methods could be used for surface kinetics and thermo, but checks for thermodynamic consistency should be included. You might find some good ideas from CVD resources/literature, and I think it might be easiest to start with just one aspect of heterogenous chemistry, like adsorption.

**Comment by bslakman** [comment]
Requirements

* A different type of reactor should be implemented in RMG besides simple, liquid, since the reaction rates are mathematically different ([=] moles/cm^2/s). Rate expressions must include:
    * Fraction of vacant sites on surface ϴv
    * Sticking coefficient between species i and site j, σij
    * Ensure mathematical consistency between gas and surface rate equations (unit tests!)
* What are i and j? For now, choose small sets of each to test concept.
    * i: SiH4, SiH3, SiH2, SiH, H, H2
    * j: Two types of Si surface sites: H-terminated or dangling bond?
* Thermo: Need chemisorption energies Qss
    * Create database of Qss for each species
    * Or use QA, atomic chemisorption energy, in known linear energy correlation
    * Check for consistency of group additive methods (unit tests!)
* Kinetics: Need database of parameters (i.e. A, EA, n) as well as database of  σij

I would appreciate any input on these, and I will have progress on a separate branch- feel free to follow.

**Comment by mliu49** [comment]
Officially in RMG now!
